leader-election-impl/src/main/java/com/palantir/paxos/SqlitePaxosStateLog.java

```diff
@@ -138,7 +138,10 @@ <V extends Persistable & Versionable> boolean[] writeBatchOfRounds(
                 @Bind("useCase") String useCase,
                 @BindPojo("round") Iterable<PaxosRound<V>> rounds);
 
-        @SqlQuery("SELECT DISTINCT(namespace) FROM paxosLog")
-        Set<String> getAllNamespaces();
+        @SqlQuery("SELECT MIN(namespace) FROM paxosLog")
+        Optional<String> getSmallestNamespace();
+
+        @SqlQuery("SELECT MIN(namespace) FROM paxosLog WHERE namespace > :namespace")
+        Optional<String> getNextSmallestNamespace(@Bind("namespace") String lastReadNamespace);
```

Comment:
Might be contentious, but we should test that the query plan has the word search and not scan

(or even match it directly)

Reply:
So getting the query plan in code is a little involved. Just adding a comment instead of a test if that's good enough?